app/frontend/wallet/account.ts

```diff
@@ -330,20 +330,46 @@ const Account = ({config, cryptoProvider, blockchainExplorer, accountIndex}: Acc
   }
 
   async function getAccountInfo(validStakepoolDataProvider: StakepoolDataProvider) {
-    const accountXpubs = await getAccountXpubs()
-    const stakingXpub = await getStakingXpub(cryptoProvider, accountIndex)
-    const stakingAddress = await myAddresses.getStakingAddress()
-    const {baseAddressBalance, nonStakingBalance, balance, tokenBalance} = await getBalance()
-    const utxos = await getUtxos()
-    const shelleyAccountInfo = await getStakingInfo(validStakepoolDataProvider)
-    const stakingHistory = await getStakingHistory(validStakepoolDataProvider)
-    const visibleAddresses = await getVisibleAddresses()
-    const transactionHistory = await getTxHistory()
-    const poolRecommendation = await getPoolRecommendation(
-      shelleyAccountInfo.delegation,
-      baseAddressBalance
-    )
-    const isUsed = await isAccountUsed()
+    const promiseBatch1 = (async () => {
```

Comment:
can you clarify what is the idea behind these separate batches? Why can't all the data be fetched in the single batch, provided `Promise.all` is called on the two batches anyway?

Reply:
Refactored.